### PR TITLE
Update optitype recipe

### PR DESCRIPTION
**Pull request by naumenko-sa**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by naumenko-sa** [comment]
@BiocondaBot please add label

**Comment by jj-umn** [comment]
@BiocondaBot please merge

**Comment by jj-umn** [comment]
@naumenko-sa  I'm familiar with BiocondaBot upload errors.  You'll probably need to ask someone else to help with that.   Thanks for updating this recipe.

**Comment by jj-umn** [comment]
@BiocondaBot please merge

**Comment by bgruening** [comment]
@BiocondaBot please merge

**Comment by bgruening** [comment]
This is still only working for admins. We need to change this.

**Comment by bgruening** [comment]
This is great, it seems its only this package :(

File: recipes/optitype/meta.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ requirements:
   run:
     - python <3
     - numpy =1.15.4
```
Comment (bgruening) [inline recipes/optitype/meta.yaml:22]: Do you need the strict numpy pinning here?

**Comment by jj-umn** [comment]
@bgruening @naumenko-sa   I'll experiment with the requirements in the recipe.

File: recipes/optitype/meta.yaml (hunk 1 above)
Comment (naumenko-sa) [reply recipes/optitype/meta.yaml:22]: A good point, I think we can relax it, I've just ran a test with numpy 1.16.6 - it works
attn @andras86 please correct if the pinning is crucial

**Comment by jj-umn** [comment]
@naumenko-sa    Why don't you try with numpy >=1.16
I'm try to get the recipe to build with python3, but getting a load of conflicts.

**Comment by naumenko-sa** [comment]
I have tested with numpy1.1.16, so all good to remove the pinning. 
Agree with @jj-umn : in our experience in bcbio OptiType works well with python2, 
and I don't think python3 transition is completed.
Please merge.

**Comment by andras86** [comment]
@bgruening @naumenko-sa The python 2.7/3.6/3.7/3.8 travis builds all pass: https://travis-ci.org/github/FRED-2/OptiType/builds/671638289 Can you spot anything unusual in the travis config perhaps?

**Comment by bgruening** [comment]
@BiocondaBot please fetch artifacts

**Comment by bgruening** [comment]
python2 is dead end, we will skip is soon on bioconda as well. So better to also build it with py3k if possible.

File: recipes/optitype/meta.yaml [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
 package:
   name: optitype
   version: {{ version }}
 
 build:
   noarch: python
-  number: 4
+  number: 0
   script: {{ PYTHON }} -m pip install . --ignore-installed --no-deps -vv
```
Comment (bgruening) [inline recipes/optitype/meta.yaml:10]: I'm not sure how this ever worked. Its not a python package with a setup.py file?
Maybe you need to copy the few python scripts into ${PREFIX}/bin ?

**Comment by dpryan79** [comment]
The bot upload errors usually only happen when the docker container is quite large. The connection times out then.

**Comment by naumenko-sa** [comment]
Since the package author @andras86 confirms that Optitype works with python3, 
and I have also tested it in the python3 conda enviroment, it works for both DNA and RNA test,
I suggest we switch OptiType recipe to python3 now in bioconda, I've updated the recipe.

@bgruening, yes, OptiType is not a python package with setup.py. The recipe installs all necessary packages, and then we run it as a command: https://github.com/bcbio/bcbio-nextgen/blob/2bcb644c1bd061add6110d483af69f14850630bc/bcbio/hla/optitype.py

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by dpryan79** [comment]
Someone needs to add a proper shebang (`#!/usr/bin/env python`) to the various python files.

**Comment by naumenko-sa** [comment]
thanks @dpryan79 
@andras86, could you please merge and cut a new release? I will then update a recipe.
https://github.com/FRED-2/OptiType/pull/106

**Comment by naumenko-sa** [comment]
@BiocondaBot please add label

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**